Added explanations + Qs for Ch2

mdncs opened the pull request:
I've added some questions for UpGoing/ch2.js and the explanations for them. Also added some links to the chapters; haven't figured out how to link to the exact line yet. Issue  #1 for Qs & #14 for explanations

nikrb commented on src/components/ChapterRouter.js:
```diff
@@ -51,6 +51,10 @@ class ChapterRouter extends Component {
         >
           <h3 style={{ fontSize: '24px' }}>{chapter.title}</h3>
+          <h6>
+            Full explanations here:
```
shouldn't the actual explanation be here, e.g. `{chapter.explanation}`

nikrb commented:
great! more questions :) 

chapter 2: is question 4 a duplicate of question 2?  *haven't checked through them all*

ch2/q2 explanation spills over question box. I thought the box looked a bit odd with all that empty space at the bottom when the question first appears. Perhaps we could display the explanation as a popup/overlay? @austintackaberry maybe create a new issue for this after merging this pr?

possibly nitpicking, but branch name could be more descriptive, e.g. instead of `master`, `feat/add-explanations`. Also possibly we have two pr's here, one for chapter 1 and a second for chapter 2? I think it's ok for now just for future reference ... try to make pr's as small (few files) as possible.

mdncs commented:
thanks for the feedback nik! you're right, I removed the question there. I left out the last part of the chapter since they're more complex concepts and have their own chapters, but let me know what you think and I could maybe pull a few more basic questions on those.

I changed the 'full explanation here' bit - it's not supposed to be the actual explanation, but link out to the actual YDKJS chapter for more info (need to figure out how to link the actual line as @austintackaberry suggested)

maybe a smaller font size for the explanations? I imagine it'll have to be mobile responsive ultimately so the layout might end up changing either way.

re the branch name and pr size - noted, thanks! it's literally the very first pr I've made so apologies for that, will keep in mind in the future :)

nikrb commented:
be great if you could add more questions. No need to apologise, I'm no expert, we all help each other out :)
